### PR TITLE
fix(remix): bump Remix to version 2

### PR DESCRIPTION
**Pull request by richard-paystack**
# Pull request template

## Describe the PR
- This PR serves primarily to update our remix version to V2. It cannibalises many of the changes suggested in t[he original stacks PR](https://github.com/remix-run/indie-stack/pull/257).
- In terms of [Stitches](https://github.com/remix-run/examples/tree/main/stitches), it adds one missing file: `stitches.config.ts` which `Keeps the Stitches configuration that is shared into the project.`
- [Link to the issue that requires this PR](https://paystack.atlassian.net/browse/CS-384)
- Shouldnt be a breaking change, we create repos from here but make sure node is updated

## Sign off

- [x] Code conforms to the [Frontend Coding Style Guide](https://www.notion.so/paystack/Frontend-Coding-Style-Guide-5918f0348a514e57a742727c164ddb6b)
- [ ] Added test(s)
- [ ] Merged this branch into `dev`
- [ ] Added sufficient inline comments and/or added to DEV_DOCS.md if needed
- [ ] Can be safely merged to master and deployed to production pending pr approval

### CODE REVIEW

File: app/styles/stitches.config.ts
Comment (alex-paystack) [inline app/styles/stitches.config.ts:1]: We should not need this. We will be using the stitches instance from Pax

File: cypress/tsconfig.json [hunk 1]
```diff
@@ -15,8 +15,8 @@
     "types": ["node", "cypress", "@testing-library/cypress"],
     "esModuleInterop": true,
     "jsx": "react-jsx",
-    "moduleResolution": "node",
-    "target": "es2019",
+    "moduleResolution": "Bundler",
```
Comment (alex-paystack) [inline cypress/tsconfig.json]: `Bundler` is better suited to packages no?

File: remix.init/index.js [hunk 2]
```diff
@@ -105,71 +78,68 @@ const updatePackageJson = ({ APP_NAME, isTypeScript, packageJson }) => {
...
-    rootDirectory,
-    ".github",
-    "workflows",
-    "deploy.yml"
-  );
-  const CYPRESS_SUPPORT_PATH = path.join(rootDirectory, "cypress", "support");
-  const CYPRESS_COMMANDS_PATH = path.join(
+
+  const README_PATH = path.join(rootDirectory, 'README.md');
+  const FLY_TOML_PATH = path.join(rootDirectory, 'fly.toml');
```
Comment (alex-paystack) [inline remix.init/index.js]: We will not be using fly for deployment so you can remove everything related to that. Also scan to see other tools and processes we will not be using and remove them from here

Comment (richard-paystack) [reply remix.init/index.js]: Could you please give this one a quick cursory glance, ive gone ahead and removed a lot of the things you originally removed as part of your stack trim PR but i might have missed a few things 🙏🏼

File: remix.init/index.js [hunk 3]
```diff
@@ -1,175 +1,119 @@
...
-  ]);
+    .replace(/[^a-zA-Z0-9-_]/g, '-');
+
+  const [prodContent, readme, env, cypressCommands, packageJson] =
+    await Promise.all([
+      fs.readFile(README_PATH, 'utf-8'),
+      fs.readFile(EXAMPLE_ENV_PATH, 'utf-8'),
+      fs.readFile(CYPRESS_COMMANDS_PATH, 'utf-8'),
+      PackageJson.load(rootDirectory),
+    ]);
```
Comment (alex-paystack) [inline remix.init/index.js]: `prodContent` should go

File: remix.init/index.js [hunk 4]
```diff
@@ -1,175 +1,119 @@
...
 
   const newEnv = env.replace(
     /^SESSION_SECRET=.*$/m,
-    `SESSION_SECRET="${getRandomString(16)}"`
+    `SESSION_SECRET="${getRandomString(16)}"`,
   );
 
+  const prodToml = toml.parse(prodContent);
+  prodToml.app = prodToml.app.replace(REPLACER, APP_NAME);
```
Comment (alex-paystack) [inline remix.init/index.js]: We still have some `toml` stuff that can go

File: remix.init/index.js [hunk 5]
```diff
@@ -1,175 +1,119 @@
...
-const path = require("path");
+const { execSync } = require('node:child_process');
+const crypto = require('node:crypto');
+const fs = require('node:fs/promises');
+const path = require('node:path');
 
-const PackageJson = require("@npmcli/package-json");
-const semver = require("semver");
-const YAML = require("yaml");
+const toml = require('@iarna/toml');
```
Comment (alex-paystack) [inline remix.init/index.js]: this can go

**Review by ademola-paystack — APPROVED** [review]

**Review by JenniferW-paystack — APPROVED** [review]

**Pull request merged**